apps/web/components/dashboard/BulkBookmarksAction.tsx

```diff
@@ -0,0 +1,204 @@
+    });
+  };
+
+  const deleteBookmarkMutator = useDeleteBookmark({
+    onSuccess: () => {
+      handleBulkEdit(false);
+    },
+    onError,
+  });
+
+  const updateBookmarkMutator = useUpdateBookmark({
+    onSuccess: () => {
+      handleBulkEdit(false);
+    },
+    onError,
+  });
+
+  interface UpdateBookmarkProps {
+    favourited?: boolean;
+    archived?: boolean;
+  }
+
+  const updateBookmarks = ({
+    favourited,
+    archived,
+  }: UpdateBookmarkProps): void => {
+    selectedBookmarks.map((item) => {
+      updateBookmarkMutator.mutate({
+        bookmarkId: item.id,
+        favourited: favourited ?? item.favourited,
+        archived: archived ?? item.archived,
+      });
+    });
+    toast({
+      description: `${selectedBookmarks.length} bookmarks have been updated!`,
+    });
+  };
+
+  const deleteBookmarks = () => {
+    selectedBookmarks.map((item) => {
+      deleteBookmarkMutator.mutate({ bookmarkId: item.id });
+    });
+    toast({
+      description: `${selectedBookmarks.length} bookmarks have been deleted!`,
+    });
+  };
+
+  const alreadyFavourited =
+    (selectedBookmarks.length &&
+      selectedBookmarks.every((item) => item.favourited === true)) ||
+    pathname.includes("favourites");
+
+  const alreadyArchived =
+    (selectedBookmarks.length &&
+      selectedBookmarks.every((item) => item.archived === true)) ||
+    pathname.includes("archive");
+
+  const actionList = [
+    {
+      name: alreadyFavourited ? "Unfavourite" : "Favourite",
+      icon: (
+        <Star
+          color={alreadyFavourited ? "#ebb434" : "#000"}
+          fill={alreadyFavourited ? "#ebb434" : "transparent"}
+          size={18}
+        />
+      ),
+      action: () => updateBookmarks({ favourited: !alreadyFavourited }),
+    },
+    {
+      name: alreadyArchived ? "Un-arhcive" : "Archive",
+      icon: alreadyArchived ? (
+        <ArchiveRestore size={18} />
+      ) : (
+        <Archive size={18} />
+      ),
+      action: () => updateBookmarks({ archived: !alreadyArchived }),
+    },
+    {
+      name: "Delete",
+      icon: <Trash2 size={18} />,
+      action: () => setShowDeleteModal(true),
+    },
+    {
+      name: "Close bulk edit",
+      icon: <X size={18} />,
+      action: () => handleBulkEdit(false),
+    },
+  ];
+
+  if (!isBulkEditEnabled) {
+    return (
+      <div>
+        <ButtonWithTooltip
+          tooltip="Bulk Edit"
+          delayDuration={100}
+          variant="ghost"
+          onClick={() => handleBulkEdit(true)}
+        >
+          <Pencil size={18} />
+        </ButtonWithTooltip>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      {showDeleteModal && (
+        <DeleteModal
+          open={showDeleteModal}
+          setOpen={setShowDeleteModal}
+          deleteBookmarks={deleteBookmarks}
+        />
+      )}
+      <div className="flex">
+        {actionList.map(({ name, icon: Icon, action }) => (
+          <ButtonWithTooltip
```

Comment:
Also, you probably want to differentiate between buttons that does actions and have loading states, and buttons that just close bulk edit. For the ones that do actions, you probably want to use `ActionButtonWithTooltip` so that they are disabled in demo mode.

Reply:
- Replace with `ActionButtonWithTooltip `
- Added `disbale` when nothing is selected